### PR TITLE
web: move code-intel bottom panel styles out from the `.theme-redesign`

### PR DESCRIPTION
**Pull request by valerybugakov**
## Changes

- Moved `Panel` styles out from the `.theme-redesign`.
- Converted `Panel` styles to CSS modules.
- Converted `Resizable` component styles to CSS modules to avoid specificity issues with `Panel` styles.

Part of the https://github.com/sourcegraph/sourcegraph/issues/20847.

### CODE REVIEW

**Comment by pdubroy** [comment]
@valerybugakov This PRs is quite a bit bigger than our [400-line guideline](https://about.sourcegraph.com/handbook/engineering/developer-insights#prefer-small-prs-lines) and it looks like it mixes several different concerns. Is it possible to break it up into multiple PRs?

File: client/branded/src/components/panel/Panel.module.scss [hunk 1]
```diff
@@ -0,0 +1,38 @@
+.resizable-panel {
```
Comment (eseliger) [inline client/branded/src/components/panel/Panel.module.scss:1]: Not related to this PR specifically, but. I wanted to say how much I enjoy how much simpler our CSS got by using CSS modules. 🚀

**Review by eseliger — APPROVED** [review]
Nice, so much simpler :)

**Comment by valerybugakov** [comment]
@pdubroy, I've extracted `HierarchicalLocationsView` cleanup in [a separate PR](https://github.com/sourcegraph/sourcegraph/pull/22008) 👍

**Review by eseliger — APPROVED** [review]

**Pull request merged**